CMakeLists.txt

```diff
@@ -257,128 +274,56 @@ endif()
 #   ENSMALLEN_INCLUDE_DIR - include directory for ensmallen
 #   STB_IMAGE_INCLUDE_DIR - include directory for STB image library
 #   MATHJAX_ROOT - root of MathJax installation
-find_package(Armadillo "${ARMADILLO_VERSION}" REQUIRED)
 
+# Remove support for older 32bit system, if we need to have 32bit system
+# Since we are cross compiling, We will download it into the deps folder,
+find_package(Armadillo "${ARMADILLO_VERSION}")
+if(NOT ARMADILLO_FOUND)
```

Comment:
I think we also need to consider the `DISABLE_DOWNLOADS` variable here---if the user told us they don't want to download anything if it's not available, we should respect that. :+1: